tasks/setup.yaml

```diff
@@ -57,36 +57,100 @@ tasks:
+          PASSWORD=$(./uds zarf tools kubectl get secret keycloak-admin-password -n keycloak -o jsonpath='{.data.password}' | base64 --decode)
 
-  - name: create-doug-user
-    description: Creates a user named 'doug' in the uds realm of keycloak (using the default admin account)
+          printf "\033[31m!!! Please ensure you're not running this in CI !!!\033[0m\n"
+          printf "Keycloak Admin Username: \033[31m %s \033[0m\n" "$USERNAME"
+          printf "Keycloak Admin Password: \033[31m %s \033[0m\n" "$PASSWORD"
+
+  - name: keycloak-admin-user
+    description: Sets up the Keycloak admin user for dev/testing if not already created
+    actions:
+      - description: Create Keycloak Admin User
+        cmd: |
+          # Check if the secret exists
+          if ./uds zarf tools kubectl get secret keycloak-admin-password -n keycloak > /dev/null 2>&1; then
+            echo "Admin user exists, skipping..."
+          else
+            # Start port-forward with zarf
+            ./uds zarf tools kubectl port-forward -n keycloak svc/keycloak-http 8080:8080 &
+            PF_PID=$!
+
+            # Wait a bit to ensure port-forward is ready
+            sleep 5
+
+            # Create admin user with curl
+            PASSWORD=$(openssl rand -base64 12)
+            STATE_COOKIE=$(curl -s --output /dev/null --cookie-jar - http://localhost:8080/ | grep "WELCOME_STATE_CHECKER" | awk '{print $7}')
+            curl -s --show-error http://localhost:8080/ \
+              -H "Cookie: WELCOME_STATE_CHECKER=${STATE_COOKIE}" \
+              -H "Content-Type: application/x-www-form-urlencoded" \
+              --data-urlencode "username=admin" \
+              --data-urlencode "password=${PASSWORD}" \
+              --data-urlencode "passwordConfirmation=${PASSWORD}" \
+              --data-urlencode "stateChecker=${STATE_COOKIE}"
+
+            # Kill the port-forward
+            kill $PF_PID
+
+            ./uds zarf tools kubectl create secret generic keycloak-admin-password \
+              --from-literal=username=admin \
+              --from-literal=password="${PASSWORD}" \
+              -n keycloak
+          fi
+
+  - name: keycloak-user
+    description: Creates a Keycloak user in the UDS Realm
     inputs:
-      keycloak_group:
+      group:
         description: Group to add user to
-        default: $KEYCLOAK_GROUP
+        default: $KEYCLOAK_USER_GROUP
+        required: false
+      username:
+        description: The username of the user to create
+        default: doug
+        required: false
+      password:
+        description: The password of the user to create
+        default: unicorn123!@#UN
+        required: false
+      first_name:
+        description: First name of the user to create
+        default: Doug
+        required: false
+      last_name:
+        description: First name of the user to create
+        default: Doug
```

Comment:
extremely nit: `doug` user's last name was Unicorn previously. I highly doubt anything is doing validations on this, but may be good to keep the default the same just in case.

Reply:
aaah you're eagle-eyed @mjnagel ! Fixed!